apiserver/facades/client/spaces/shims.go

```diff
@@ -154,3 +159,53 @@ func (s *stateShim) ConstraintsBySpaceName(spaceName string) ([]Constraints, err
 	}
 	return cons, nil
 }
+
+type backingReloadSpacesShim struct {
+	Backing
+}
+
+func (b backingReloadSpacesShim) AllSpaces() ([]space.Space, error) {
+	spaces, err := b.Backing.AllSpaces()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	result := make([]space.Space, len(spaces))
+	for i, space := range spaces {
+		result[i] = backingSpaceShim{
+			BackingSpace: space,
+		}
+	}
+	return result, nil
+}
+
+func (b backingReloadSpacesShim) AddSpace(name string, providerId network.Id, subnetIds []string, public bool) (space.Space, error) {
+	space, err := b.Backing.AddSpace(name, providerId, subnetIds, public)
+	if err != nil {
+		return backingSpaceShim{}, errors.Trace(err)
+	}
+	return backingSpaceShim{
+		BackingSpace: space,
+	}, nil
+}
+
+func (b backingReloadSpacesShim) ConstraintsBySpaceName(spaceName string) ([]space.Constraints, error) {
+	constraints, err := b.Backing.ConstraintsBySpaceName(spaceName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	result := make([]space.Constraints, len(constraints))
+	for i, constraint := range constraints {
+		result[i] = constraint
+	}
+	return result, nil
+}
+
+type backingSpaceShim struct {
+	networkingcommon.BackingSpace
```

Comment:
Not right now, because we don't return structs from our interface abstractions. There is no `state.Space`, only `networkingcommon.BackingSpace`. So you need to shim that to get it into `space.Space`. Turtles all the way down.

One possibility is that we return `state.Space`, but then that means it's difficult to test and why we do this indirection.